modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java

```diff
@@ -696,7 +695,22 @@ public Publisher<BinaryRow> scan(int p, @Nullable InternalTransaction tx) {
             );
         }
 
-        return new PartitionScanPublisher((scanId, batchSize) -> enlistCursorInTx(tx, p, scanId, batchSize));
+        final boolean implicit = tx == null;
+
+        final InternalTransaction tx0 = implicit ? txManager.begin() : tx;
+
+        return new PartitionScanPublisher(
+                (scanId, batchSize) -> enlistCursorInTx(tx0, p, scanId, batchSize),
+                () -> {
+                    if (implicit) {
+                        try {
+                            tx0.commitAsync().get();
```

Comment:
Nope, we should use common postEnlist because it may contain lot's of useful stuff besides commitAsync()
It's not valid to call get(), both Publisher and postEnlist have async nature, use java.util.concurrent.Flow.Subscriber#onComplete in case of successful result and java.util.concurrent.Flow.Subscriber#onError otherise.